src/routes/connect/Connect.svelte

```diff
@@ -1,28 +1,34 @@
 <script lang="ts">
+    import Albedo from './ui/wallets/Albedo';
+    import Freighter from './ui/wallets/Freighter';
+    import XBull from './ui/wallets/XBull';
+    import PrivateKey from './ui/wallets/PrivateKey';
     import { onMount } from 'svelte';
-    import { isPrivateKeyVisible, inputValue, connectionError } from './connectStore';
-    import { encryptPrivateKey, getStellarKeypair, decryptPrivatePair } from './connectHelpers';
-    import { publicKey } from '../../store/store';
-    import InvalidPrivateKeyError from './errors/InvalidPrivateKeyError';
+    import { inputValue, isPrivateKeyVisible, isWalletHidden } from './connectStore';
+    import { decryptPrivatePair } from './connectHelpers';
     import StorageKeyNotFoundError from './errors/StorageKeyNotFoundError';
+    import { albedo, xBull, freighter, privateKey } from '../../assets/index';
 
-    async function connectWithSecretKey(key: string): Promise<void> {
-        try {
-            const stellarKeyPair = await getStellarKeypair(key);
-            encryptPrivateKey(key);
-            $publicKey = stellarKeyPair.publicKey();
-        } catch (e) {
-            if (e instanceof InvalidPrivateKeyError) {
-                $publicKey = '';
-                $connectionError = 'Invalid key, please try again';
-            }
-        }
+    async function connectWithAlbedo() {
+        return new Albedo().logIn();
+    }
+
+    async function connectWithFreighter() {
+        return new Freighter().logIn();
+    }
+
+    async function connectWithXBull() {
+        return new XBull().logIn();
+    }
+
+    async function connectWithSecretKey(privateKey: string): Promise<void> {
+        return new PrivateKey().logIn(privateKey);
     }
 
     onMount(async function connectWithStorage(): Promise<void> {
```

Comment:
I'm not sure if the `onMount` is needed

Reply:
Actually, we need to improve this, by changing the button of "connect" to a text "connected with wallet" when there is a key in the storage, also adding a button to connect with a different wallet